src/Microsoft.Azure.WebJobs.Host/Bindings/BindingFactory.cs

```diff
@@ -136,60 +136,60 @@ public IBindingProvider BindToGenericValueProvider<TAttribute>(Func<TAttribute,
 
             return new GenericAsyncCollectorBindingProvider<TAttribute>(this._nameResolver, builder, filter);
         }
-        
-        /// <summary>
-        /// Create a binding provider that binds to the user parameter type. This skips the converter manager. 
-        /// </summary>
-        /// <param name="builder">Builder function that takes (resolved attribute, user parameter type) and returns an object that is assigned to the user parameter type.</param>
-        /// <returns>A binding provider that applies these semantics.</returns>
-        public IBindingProvider BindToGenericItem<TAttribute>(Func<TAttribute, Type, Task<object>> builder)
-            where TAttribute : Attribute
-        {
-            return new GenericItemBindingProvider<TAttribute>(builder, this._nameResolver);
-        }
 
         /// <summary>
-        /// Create a binding provider that binds the parameter to an specific instance of TUserType. 
+        /// General rule for binding to an input type for a given attribute. 
         /// </summary>
-        /// <typeparam name="TAttribute">type of binding attribute on the user's parameter.</typeparam>
-        /// <typeparam name="TUserType">The exact type of the user's parameter that this will bind to.</typeparam>
-        /// <param name="buildFromAttribute">builder function to create the object that will get passed to the user function.</param>
-        /// <returns>A binding provider that applies these semantics.</returns>
-        public IBindingProvider BindToExactType<TAttribute, TUserType>(Func<TAttribute, TUserType> buildFromAttribute)
-            where TAttribute : Attribute
+        /// <typeparam name="TAttribute">Type of binding attribute on the user's parameter.</typeparam>
+        /// <typeparam name="TType">The user type must be compatible with this type for the binding to apply.</typeparam>
+        /// <param name="allowConversions">
+        /// If false, then the user parameter type must exactly match TType. 
+        /// If true, this rule can use the ConverterManager to convert from TType to the final parameter type </param>
+        /// <param name="builderType">A class with builder methods on it. This will get instantiated to perform the builder rule.</param>
+        /// <param name="constructorArgs">constructor arguments to pass to the typeBuilder instantiation. This can be used 
+        /// to flow state (like configuration, secrets, etc) from the configuration to the specific binding</param>
+        /// <returns>A binding rule.</returns>
+        public IBindingProvider BindToInput<TAttribute, TType>(
+            bool allowConversions,
+            Type builderType,
```

Comment:
`BindToInput` changed to using a builder Type (which I'm fine with -- it may even make things nicer), but why didn't the other `BindTo` methods change? AsyncCollector and ValueProvider should match this pattern. If I'm writing a new binding, I'd expect the concepts to be similar for each rule I add.